test_util/test_azure.py

```diff
@@ -0,0 +1,141 @@
+    py.test -vv CI_FLAGS integration_test.py
+
+TEST_ADD_ENV_*: string (default=None)
+    Any number of environment variables can be passed to integration_test.py if
+    prefixed with 'TEST_ADD_ENV_'. The prefix will be removed before passing
+"""
+import logging
+import os
+import sys
+
+from contextlib import ExitStack
+
+from gen.calc import calculate_environment_variable
+from ssh.tunnel import Tunnel
+from test_util.azure import AzureWrapper, TemplateDeployment
+from test_util.helpers import random_id
+from test_util.test_runner import integration_test
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+
+def check_environment():
+    """Test uses environment variables to play nicely with TeamCity config templates
+    Grab all the environment variables here to avoid setting params all over
+
+    Returns:
+        object: generic object used for cleanly passing options through the test
+
+    Raises:
+        AssertionError: if any environment variables or resources are missing
+            or do not conform
+    """
+    options = type('Options', (object,), {})()
+
+    # Required
+    options.public_ssh_key = calculate_environment_variable('AZURE_PUBLIC_SSH_KEY')
+    options.subscription_id = calculate_environment_variable('AZURE_SUBSCRIPTION_ID')
+    options.tenant_id = calculate_environment_variable('AZURE_TENANT_ID')
+    options.client_id = calculate_environment_variable('AZURE_CLIENT_ID')
+    options.client_secret = calculate_environment_variable('AZURE_CLIENT_SECRET')
+    options.template_url = calculate_environment_variable('DCOS_TEMPLATE_URL')
+
+    # Provided if not set
+    options.name = os.getenv('DCOS_NAME', 'testing-{}'.format(random_id(10)))
+    options.ssh_key_path = os.getenv('DCOS_SSH_KEY_PATH', 'ssh_key')
+    options.location = os.getenv('AZURE_LOCATION', 'East US')
+    options.linux_user = os.getenv('AZURE_LINUX_USER', 'dcos')
+    # Prefixes must not begin with a number
+    options.agent_prefix = os.getenv('AZURE_AGENT_PREFIX', 'test' + random_id(10).lower())
+    options.master_prefix = os.getenv('AZURE_MASTER_PREFIX', 'test' + random_id(10).lower())
+    options.vm_size = os.getenv('AZURE_VM_SIZE', 'Standard_D2')
+    options.num_agents = int(os.getenv('AGENTS', '2'))
+    options.name_suffix = os.getenv('AZURE_DCOS_SUFFIX', '12345')
+    options.oauth_enabled = os.getenv('AZURE_OAUTH_ENABLED', 'false') == 'true'
+    options.vm_diagnostics_enabled = os.getenv('AZURE_VM_DIAGNOSTICS_ENABLED', 'true') == 'true'
+    options.azure_cleanup = os.getenv('AZURE_CLEANUP', 'true') == 'true'
+    options.ci_flags = os.getenv('CI_FLAGS', '')
+
+    add_env = {}
+    prefix = 'TEST_ADD_ENV_'
+    for k, v in os.environ.items():
+        if k.startswith(prefix):
+            add_env[k.replace(prefix, '')] = v
+    options.add_env = add_env
```

Comment:
I really dislike passing this explicitly at all. I would prefer if this is what we use instead: https://github.com/dcos/dcos/pull/989